src/EditorFeatures/CSharpTest/ConvertToRecord/ConvertToRecordTests.cs

```diff
@@ -3988,6 +3988,265 @@ public record C(int P, bool B);
+    {
+        public static C GetC()
+        {
+            return new C
+            {
+                P = 0,
+            };
+        }
+    }
+}
+";
+            var changedMarkup = @"
+namespace N
+{
+    public record C(int P, bool B);
+
+    public static class D
+    {
+        public static C GetC()
+        {
+            return new C(0, default);
+        }
+    }
+}
+";
+            await TestRefactoringAsync(initialMarkup, changedMarkup).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestMovePropertiesAndDontRefactorInitializerWithExistingConstructor()
+        {
+            var initialMarkup = @"
+namespace N
+{
+    public class [|C|]
+    {
+        public int P { get; init; }
+        public bool B { get; init; }
+
+        public C(int p)
+        {
+            P = p;
+            B = false;
+        }
+    }
+
+    public static class D
+    {
+        public static C GetC()
+        {
+            return new C(0)
+            {
+                B = true,
+            };
+        }
+    }
+}
+";
+            var changedMarkup = @"
+namespace N
+{
+    public record C(int P, bool B)
+    {
+        public C(int p) : this(p, false)
+        {
+        }
+    }
+
+    public static class D
+    {
+        public static C GetC()
+        {
+            return new C(0)
+            {
+                B = true,
+            };
+        }
+    }
+}
+";
+            await TestRefactoringAsync(initialMarkup, changedMarkup).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestMovePropertiesAndRefactorInitializerInSeparateFile()
+        {
+            var initialMarkup1 = @"
+namespace N
+{
+    public class [|C|]
+    {
+        public int P { get; init; }
+        public bool B { get; init; }
+    }
+}
+";
+            var initialMarkup2 = @"
+using N;
+
+namespace N2
+{
+    public static class D
+    {
+        public static C GetC()
+        {
+            return new C
+            {
+                P = 0,
+                B = true,
+            };
+        }
+    }
+}";
+            var changedMarkup1 = @"
+namespace N
+{
+    public record C(int P, bool B);
+}
+";
+            var changedMarkup2 = @"
+using N;
+
+namespace N2
+{
+    public static class D
+    {
+        public static C GetC()
+        {
+            return new C(0, true);
+        }
+    }
+}";
+            await new RefactoringTest
+            {
+                TestState =
+                {
+                    Sources =
+                    {
+                        initialMarkup1,
+                        initialMarkup2
+                    }
+                },
+                FixedState =
+                {
+                    Sources =
+                    {
+                        changedMarkup1,
+                        changedMarkup2
+                    }
+                }
+            }.RunAsync().ConfigureAwait(false);
+        }
+
         #region selection
```

Comment:
Do we have a test for nested initializers? i.e. `new C { Prop = new Foo { Prop = null } }`